hazelcast/src/test/java/com/hazelcast/cache/impl/InternalCacheRecordStoreTest.java

```diff
@@ -64,17 +88,114 @@ public void testBatchEventMapShouldBeCleanedAfterRemoveAll() {
         assertEquals(0, recordStore.batchEvent.size());
     }
 
-    @Override
-    protected void onSetup() {
+    /**
+     * Test for issue: https://github.com/hazelcast/hazelcast/issues/6983
+     */
+    @Test
+    public void ownerStateShouldBeUpdatedAfterMigration() throws ExecutionException, InterruptedException {
```

Comment:
if it seems possible to you, a shorter test makes much sense, this seems a hard one to understand to me.